internal/service/studies/agreements.go

```diff
@@ -0,0 +1,58 @@
+package studies
+
+import (
+	"github.com/google/uuid"
+	"github.com/ucl-arc-tre/portal/internal/config"
+	openapi "github.com/ucl-arc-tre/portal/internal/openapi/web"
+	"github.com/ucl-arc-tre/portal/internal/types"
+)
+
+// creates a signature record for a user confirming an agreement for a specific study
+func (s *Service) ConfirmStudyAgreement(user types.User, studyID uuid.UUID, agreementID uuid.UUID) error {
+	// Verify the study exists and the user has access to it
+	_, err := s.StudyWithOwner(studyID, user)
```

Comment:
I'm not sure about this mixture of RBAC. let's give it some thought

Reply:
it's time to think about the complicated thing – dynamic policies

Reply:
happy to not do it now though. however – I think the "is the user id an owner of the study" a sufficiently reused function so I'd write add and use

```go
func (s *Service) func UserOwnsStudy(user types.User, studyId uuid.UUID) (bool, error)
```